pkg/util/iptables/iptables.go

```diff
@@ -601,13 +601,17 @@ func (runner *runner) Monitor(canary Chain, tables []Table, reloadFunc func(), i
 
 // chainExists is used internally by Monitor; none of the public Interface methods can be
 // used to distinguish "chain exists" from "chain does not exist" with no side effects
+// it does not wait to acquire the iptables lock, avoiding to lock the runner unnecessarely
 func (runner *runner) chainExists(table Table, chain Chain) (bool, error) {
+	iptablesCmd := iptablesCommand(runner.protocol)
 	fullArgs := makeFullArgs(table, chain)
+	fullArgs = append([]string{string(opListChain)}, fullArgs...)
 
 	runner.mu.Lock()
```

Comment:
another  option I think may work is not using the runner lock here and rely on the iptables lock. I prefer this one, this operation should not take long now, we fail fast and still control from the runner the iptables operations. 
also the iptables-nft does not have this external lock , so don't know how this will look like in the future :shrug:

Reply:
I don't think there's any need to rewrite this function. If it takes long to run then that's a problem, and it's good that it causes problems in that case, because then we'll notice the problem and fix it.

Reply:
Actually... I don't think `iptables.runner` needs a mutex at all. None of its own fields are mutable after creation, and we don't need to serialize calls to the iptables binary, because it has its own lock.

Reply:
I left it as is, adding a trace so it logs the time to execute